hazelcast/src/test/java/com/hazelcast/spi/impl/merge/AbstractContainerMergerTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.merge;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.RequireAssertEnabled;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.spi.impl.merge.TestMergeOperation.OperationMode.BLOCKS;
+import static com.hazelcast.spi.impl.merge.TestMergeOperation.OperationMode.THROWS_EXCEPTION;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class AbstractContainerMergerTest extends HazelcastTestSupport {
+
+    private NodeEngineImpl nodeEngine;
+    private TestContainerCollector collector;
+    private TestContainerCollector emptyCollector;
+
+    @Before
+    public void setUp() {
+        HazelcastInstance hazelcastInstance = createHazelcastInstance();
+        nodeEngine = getNodeEngineImpl(hazelcastInstance);
+        warmUpPartitions(hazelcastInstance);
+
+        collector = new TestContainerCollector(nodeEngine, true, true);
+        collector.run();
+
+        emptyCollector = new TestContainerCollector(nodeEngine, false, true);
+        emptyCollector.run();
+    }
+
+    /**
+     * Tests that the merger finished under normal conditions.
+     */
+    @Test
+    @RequireAssertEnabled
```

Comment:
Why is this necessary? Also, in order to be taken into account, it seems a `@Rule` is missing:
```
 * To have this annotation honored, a test class must define an instance of {@link AssertEnabledFilterRule}.
 * This is automatically done via {@link AutoRegisteredTestRule}.
```

Reply:
It's a safety net to ensure that asserts are enabled (it will let the test fail if they are not working). It's also a nice marker for a test, that it tests an `assert` in the production code, so most probably here will be no `assertEquals()` or so in the test code.

The rule is automatically added via `AutoRegisteredTestRule`. The JavaDoc maybe a little misleading here.